src/System.Data.Common/src/System/Data/Common/DbCommand.cs

```diff
@@ -175,9 +178,12 @@ protected virtual Task<DbDataReader> ExecuteDbDataReaderAsync(CommandBehavior be
                 }
                 catch (Exception e)
                 {
-                    registration.Dispose();
                     return Task.FromException<DbDataReader>(e);
                 }
+                finally
+                {
+                    registration.Dispose();
```

Comment:
There's no asynchrony here... the try block is calling the synchronous ExecuteReader and wrapping the result in a task.